screens/AppNavigator/screens/BalancesScreen/BalancesScreen.tsx

```diff
@@ -1,52 +1,67 @@
+import { useDispatch, useSelector } from 'react-redux'
 import tailwind from 'tailwind-rn'
+import { NumberText } from '../../../../components'
 
 import { getTokenIcon } from '../../../../components/icons/tokens/_index'
 import { useWhaleApiClient } from '../../../../hooks/api/useWhaleApiClient'
-import { useWalletAPI } from '../../../../hooks/wallet/WalletAPI'
+import { fetchTokens, useTokensAPI } from '../../../../hooks/wallet/TokensAPI'
+import { RootState } from '../../../../store'
+import { translate } from '../../../../translations'
 
 export function BalancesScreen (): JSX.Element {
-  const whaleApiClient = useWhaleApiClient()
-  const account = useWalletAPI().getWallet().get(0)
+  const address = useSelector((state: RootState) => state.wallet.address)
+  const [refreshing, setRefreshing] = useState(false)
+  const dispatch = useDispatch()
+  const whaleAPI = useWhaleApiClient()
 
-  const [utxoBalance, setUtxoBalance] = React.useState<string>('')
-  const [tokens, setTokens] = React.useState<AddressToken[]>([])
-
-  useEffect(() => {
-    account.getAddress().then(async address => {
-      const balance = await whaleApiClient.address.getBalance(address)
-      setUtxoBalance(balance)
-
-      const tokens = await whaleApiClient.address.listToken(address, 30)
-      setTokens([...tokens])
-    }).catch(() => {
-      console.log('@typescript-eslint/no-floating-promises')
-    })
+  const onRefresh = useCallback(async () => {
+    setRefreshing(true)
+    await fetchTokens(address, dispatch, whaleAPI)
+    setRefreshing(false)
   }, [])
 
-  const Icon = getTokenIcon('DFI')
-
-  const balances = tokens.map(token => {
-    const Icon = getTokenIcon(token.symbol)
-
-    return (
-      <View key={token.id} style={tailwind('flex-row items-center')}>
-        <Icon />
-        <Text style={tailwind('ml-2')}>{token.amount}</Text>
-      </View>
-    )
-  })
+  const tokens = useTokensAPI()
+  return (
+    <View style={tailwind('items-start justify-start w-full flex-grow')}>
+      <FlatList
+        style={tailwind('w-full mb-4')}
+        data={tokens}
+        testID='balances_list'
+        refreshControl={
+          <RefreshControl
+            refreshing={refreshing}
+            onRefresh={onRefresh}
+          />
+        }
+        renderItem={({ item }) => <BalanceItemRow token={item} key={item.id} />}
+        ListHeaderComponent={
+          <Text style={tailwind('font-bold p-4 text-base')}>
+            {translate('screens/BalancesScreen', 'Portfolio')}
+          </Text>
+        }
+      />
+    </View>
+  )
+}
 
+function BalanceItemRow ({ token }: { token: AddressToken }): JSX.Element {
+  const Icon = getTokenIcon(token.symbol)
   return (
-    <View style={tailwind('flex-1 items-center justify-center')}>
-      <View style={tailwind('flex-row items-center')}>
+    <View testID={token.id} style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
```

Comment:
try to use as little style as possible as not everything is required for rendering what you want